airbyte-integrations/connectors/source-sftp/src/main/resources/spec.json

```diff
@@ -0,0 +1,122 @@
+{
+  "documentationUrl": "https://docs.airbyte.io/integrations/source/sftp",
+  "connectionSpecification": {
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "title": "SFTP Source Spec",
+    "type": "object",
+    "required": ["user", "host", "port"],
+    "additionalProperties": true,
+    "properties": {
+      "user": {
+        "title": "User Name",
+        "description": "The server user",
+        "type": "string",
+        "order": 0
+      },
+      "host": {
+        "title": "Host Address",
+        "description": "The server host address",
+        "type": "string",
+        "examples": [
+          "www.host.com",
+          "192.0.2.1"
+        ],
+        "order": 1
+      },
+      "port": {
+        "title": "Port",
+        "description": "The server port",
+        "type": "integer",
+        "default": 22,
+        "examples": [
+          "22"
+        ],
+        "order": 2
+      },
+      "credentials" : {
+        "type": "object",
+        "title": "Authentication *",
+        "description": "The server authentication method",
+        "order" : 3,
+        "oneOf": [
+          {
+            "title": "Password Authentication",
+            "required": [
+              "auth_method",
+              "auth_user_password"
+            ],
+            "properties": {
+              "auth_method": {
+                "description": "Connect through password authentication",
+                "type": "string",
+                "const": "SSH_PASSWORD_AUTH",
+                "order": 0
+              },
+              "auth_user_password": {
+                "title": "Password",
+                "description": "OS-level password for logging into the jump server host",
+                "type": "string",
+                "airbyte_secret": true,
+                "order": 1
+              }
+            }
+          },
+          {
+            "title": "SSH Key Authentication",
+            "required": [
+              "auth_method",
+              "auth_ssh_key"
+            ],
+            "properties": {
+              "auth_method": {
+                "description": "Connect through ssh key",
+                "type": "string",
+                "const": "SSH_KEY_AUTH",
+                "order": 0
+              },
+              "auth_ssh_key": {
+                "title": "SSH Private Key",
+                "description": "OS-level user account ssh key credentials in RSA PEM format ( created with ssh-keygen -t rsa -m PEM -f myuser_rsa )",
+                "type": "string",
+                "airbyte_secret": true,
+                "multiline": true,
+                "order": 1
+              }
+            }
+          }
+        ]
+      },
+      "file_type": {
+        "title": "File type",
+        "description": "The file type for data sync",
```

Comment:
If this parameter allows multiple coma separated values, then the name should be "File types" and description should be "Coma separated file types"

Reply:
Yes, sure. I guess we need to request multi select drop down from UI team to avoid this comma separated thing it's not user friendly